packages/graphiql-react/src/editor/context.tsx

```diff
@@ -1,24 +1,32 @@
-import type { Editor } from 'codemirror';
 import { createContext, ReactNode, useState } from 'react';
 
+import { CodeMirrorEditor } from './types';
+
 export type EditorContextType = {
-  headerEditor: Editor | null;
-  setHeaderEditor(newEditor: Editor): void;
+  headerEditor: CodeMirrorEditor | null;
+  queryEditor: CodeMirrorEditor | null;
+  setHeaderEditor(newEditor: CodeMirrorEditor): void;
+  setQueryEditor(newEditor: CodeMirrorEditor): void;
 };
 
 export const EditorContext = createContext<EditorContextType>({
   headerEditor: null,
+  queryEditor: null,
   setHeaderEditor() {},
+  setQueryEditor() {},
```

Comment:
Just out of curiosity - why are these empty functions?

Reply:
TypeScript wants you to provide a default value when creating a context. When actually rendering the context provider we'll pass in the actual value that contains the functions to set the editors in state.